### PR TITLE
Remove MessageEvent and make Event extensible.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/openconsensus/trace/Event.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import javax.annotation.concurrent.Immutable;
+import openconsensus.trace.data.AttributeValue;
+
+/**
+ * A text annotation with a set of attributes.
+ *
+ * @since 0.1.0
+ */
+@Immutable
+public abstract class Event {
```
Comment (SergeyKanzhelev) [inline api/src/main/java/openconsensus/trace/Event.java]: API should allow to create an Event. We don't have this option any longer, do we?

Comment (bogdandrutu) [reply api/src/main/java/openconsensus/trace/Event.java]: User can add events via the Span.event api which supports all the variants. Do we still need an event implementation here?

Comment (SergeyKanzhelev) [reply api/src/main/java/openconsensus/trace/Event.java]: How do you create event for spandata?

Comment (bogdandrutu) [reply api/src/main/java/openconsensus/trace/Event.java]: Done.

File: api/src/main/java/openconsensus/trace/ImmutableEvent.java [hunk 2]
```diff
@@ -30,7 +31,7 @@
  */
 @Immutable
 @AutoValue
-public abstract class Event {
+public abstract class ImmutableEvent extends Event {
```
Comment (SergeyKanzhelev) [inline api/src/main/java/openconsensus/trace/ImmutableEvent.java]: can this class be hidden and we will provide static `create` method on `Event`? It feels unnecessary to create another type of event.

File: api/src/main/java/openconsensus/trace/Event.java (hunk 1 above)
Comment (SergeyKanzhelev) [reply api/src/main/java/openconsensus/trace/Event.java]: See my comment. Any reason to create public type, not simply return it from the static method on `Event`?

Comment (bogdandrutu) [reply api/src/main/java/openconsensus/trace/Event.java]: Done. Let's address the interface change separately.

File: api/src/main/java/openconsensus/trace/ImmutableEvent.java (hunk 2 above)
Comment (bogdandrutu) [reply api/src/main/java/openconsensus/trace/ImmutableEvent.java]: Done. Let's address the interface change separately.

**Pull request merged**